### PR TITLE
Manage Subscription Settings

### PR DESCRIPTION
**Pull request by afterxleep**
Task/Issue URL: https://app.asana.com/0/0/1206130131426002/f

**Description**:
- Integrate Subscription management options in settings
- Adds subscription Debug options to debug menu
- Minor fixes in SwiftUI Settings
- Base Restore Flow

**Test Requirements**:
- ⚠️ Test on device
- ⚠️ Requires App Store Sandbox - 
- ⚠️ Test `using DuckDuckGo-Alpha` Target

### Test Settings Options:

1. Purchase a Subscription via the sandbox
2. Once purchased, observe that the following subscription management (See image bewlo) options are available in Settings

image

3. Tap "Manage Subscription"
4. Observe Subscription Details are shown

### Test AppStore Restore:
1. Go to Settings > All Debug Options > Subscriptions
2. Tap "Clear Authentication Data"
5. Tap "Sync App Store Account" enter the Sandbox Password and wait for confirmation
&. Observe Privacy Pro state is reset to Purchase Info as follows:

image

6. Tap Learn More and try to subscribe again while you're still subscribed.  (Subscription duration is ~5 minutes)
7. You should see a "Subscription Found" message like this one:

image

8. Tap "Restore"
9. You should be take to the subscription settings
10. Check Settings Menu, observe subscription management options are there again.
---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Core/UserDefaultsPropertyWrapper.swift [hunk 1]
```diff
@@ -119,6 +119,8 @@ public struct UserDefaultsWrapper<T> {
         case bookmarksMigrationVersion = "com.duckduckgo.ios.bookmarksMigrationVersion"
         
         case privacyConfigCustomURL = "com.duckduckgo.ios.privacyConfigCustomURL"
+        
+        case privacyProhasActiveSubscription = "com.duckduckgo.ios.privacyPro.hasActiveSubscription"
```
Comment (miasma13) [inline Core/UserDefaultsPropertyWrapper.swift]: privacyProhasActiveSubscription -> privacyPro**H**asActiveSubscription

File: DuckDuckGo/UserText.swift [hunk 2]
```diff
@@ -973,6 +973,16 @@ But if you *do* want a peek under the hood, you can find more information about
 
     public static let settingsPProLearnMore = NSLocalizedString("settings.ppro.learn.more", value: "Learn More", comment: "Learn more button text for privacy pro")
     
+    public static let settingsPProManageSubscription = NSLocalizedString("settings.ppro.manage", value: "Subscription Settings", comment: "Subscription Settings button text for privacy pro")
+    
+    public static let settingsPProVPNTitle = NSLocalizedString("settings.ppro.VPN.title", value: "VPN (Virtual Private Network", comment: "VPN cell title for privacy pro")
+    public static let settingsPProDBPTitle = NSLocalizedString("settings.ppro.DBP.title", value: "Personal Information Removal", comment: "Data Broker protection cell title for privacy pro")
+    public static let settingsPProDBPSubTitle = NSLocalizedString("settings.ppro.DBP.subtitle", value: "Remove your info from sites that sell it", comment: "Data Broker protection cell subtitle for privacy pro")
+    public static let settingsPProITRTitle = NSLocalizedString("settings.ppro.ITR.title", value: "Identity Theft Restioration", comment: "Identity theft restoration cell title for privacy pro")
```
Comment (miasma13) [inline DuckDuckGo/UserText.swift]: Typo in `Restioration`

File: DuckDuckGo/UserText.swift [hunk 3]
```diff
@@ -973,6 +973,16 @@ But if you *do* want a peek under the hood, you can find more information about
 
     public static let settingsPProLearnMore = NSLocalizedString("settings.ppro.learn.more", value: "Learn More", comment: "Learn more button text for privacy pro")
     
+    public static let settingsPProManageSubscription = NSLocalizedString("settings.ppro.manage", value: "Subscription Settings", comment: "Subscription Settings button text for privacy pro")
+    
+    public static let settingsPProVPNTitle = NSLocalizedString("settings.ppro.VPN.title", value: "VPN (Virtual Private Network", comment: "VPN cell title for privacy pro")
```
Comment (miasma13) [inline DuckDuckGo/UserText.swift]: According to designs it should be "VPN"

**Review by miasma13 — APPROVED** [review]

File: DuckDuckGo/SettingsViewModel.swift [hunk 4]
```diff
@@ -217,40 +212,64 @@ extension SettingsViewModel {
...
+                                     let syncService = legacyViewProvider.syncService
+                                     let isDataSyncingDisabled = !syncService.featureFlags.contains(.dataSyncing)
+                                      && syncService.authState == .active
+                                     if SyncBookmarksAdapter.isSyncBookmarksPaused
+                                         || SyncCredentialsAdapter.isSyncCredentialsPaused
+                                         || isDataSyncingDisabled {
+                                         return "⚠️ \(UserText.settingsSync)"
+                                     }
+                                     return UserText.settingsSync
+                                 }())
```
Comment (ayoy) [inline DuckDuckGo/SettingsViewModel.swift]: 👍

Comment (ayoy) [reply DuckDuckGo/SettingsViewModel.swift]: Please just rename `UserText.settingsSync` to `SyncUI.UserText.syncTitle` (importing SyncUI in this file as needed). UserText.settingsSync should be removed. Thanks!

**Pull request merged**